tests/Integration/Schema/Directives/WithCountDirectiveTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Tests\Integration\Schema\Directives;
+
+use Nuwave\Lighthouse\Support\AppVersion;
+use Tests\DBTestCase;
+use Tests\Utils\Models\Task;
+use Tests\Utils\Models\User;
+
+class WithCountDirectiveTest extends DBTestCase
```

Comment:
The tests in this class need to be more thorough. I don't see how they validate the directive actually eager-loads the count and saves database queries.

Also, the semantics for the `_count` convention should be defined (and tested) more clearly.

Reply:
I've added some more tests to ensure to show queries being saved.

Trying to make the semantics clearer as well. Although it would be most common, there are going to be cases where you may not even use the `_count` convention. For example:


```graphql
{
  tasks_count: Int @withCount #This is a typical case
  unread_notifications: Int @withCount(relation: "notifications", scopes: ["unread"]) # I think this would be a common case
  balance: Int @withCount(relation: "transactions", scopes: ["withBalanceSum"]) #An abstract case but plausible – altering the query constraint to get a different type of aggregate like SUM(column). I actually used something of the sort yesterday!

}
```